src/sage/matroids/graphic_matroid.pyx

```diff
@@ -277,40 +271,37 @@ def _rank(self, X):
             sage: M.rank([0,3])
             1
         """
-        from sage.sets.disjoint_set import DisjointSet
-
-        edges = self.groundset_to_edges(X)
-        vertices = set([u for (u, v, l) in edges]).union(
-            [v for (u, v, l) in edges])
+        cdef DisjointSet_of_hashables DS_vertices
+        cdef list edges = self.groundset_to_edges(X)
```

Comment:
```suggestion
        cdef list edges = self.groundset_to_edges(X)
```
Wouldn't it be better for the method to know its output type is a `list`? This applies in many places/methods. This helps Cython know to avoid the type checking when generating its C code.. Although I guess this isn't done at-large in the matroid Cython code...

Reply:
I am not sure whether it is better, but I can do it all over the matroid code.

Reply:
My understanding and intuition (i.e., without explicitly checking) is that this involves a Cython type check that the output is indeed a `list` before assigning it to `edges`. While on any individual operation, this is nominal, but in functions that can be called in tight loops, these little bits can add up and take away from the full Cython speed. Anyways, it's not something that necessarily needs to be done here.

Reply:
I see. I may do this for the whole of the `matroids` module on some moonlit night.